### ISSUE TITLE
Incorrect nullability warning on conditional expression with enum

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#26746]
@bgrainger reported this with a concise example:
https://github.com/bgrainger/NullableReferenceTypes/blob/master/BitFlags.cs

```C#
using System.IO;

namespace NullableReferenceTypes
{
	class BitFlags
	{
		public FileAccess GetFileAccess(bool isWritable)
		{
			return FileAccess.Read |
				// warning CS8626: No best nullability for operands of conditional expression 'int' and 'FileAccess'.
				(isWritable ? 0 : FileAccess.Write);

			// workaround: use 'default'
			// (isWritable ? default : FileAccess.Write);
		}
	}
}
```

I've run into this while dogfooding on Roslyn as well (`GetMarshallingType` method).

**Comment by jcouv** [comment]
Here's another repro (I think it's the same issue):
```C#
public class C 
{
    public void M() 
    {
        uint a = 0;
        uint x = true ? a : 1; // warning CS8626: No best nullability for operands of conditional expression 'uint' and 'int'.
    }
}
```
[Example](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQDk4BstWFiAJQkiQDsBjCAFQE8AHFAGhhAEssmATEAagB8AsACgAAgGYABKIBMkgMKSRAbxGT10qaIAskgLIAKAJRLhGyarPmNcNmRiSokgLySADAG4119bfuSADxdJGAQ4CEkAfkdJEEkARg9JAHpkyQB3KAQyOwBzBQBlAA4ANhkSuPQAe0lgFAcybFxgDjYYWkkwKoRJKsYEKDIuZF6wSQoqoba2SagsSQgA+iRkZBmySQByPxhNxyGtu12AOm91AF8Rc6A==)
